### PR TITLE
Fix slots, add declarative-shadow-dom if needed

### PR DESCRIPTION
**Pull request by Mihaiii**
For Starcounter/RebelsLounge#152 .

### CODE REVIEW

File: src/Products/wwwroot/Products/viewmodels/MasterPage.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <template>
     <template is="dom-bind">
         <template is="dom-if" if="{{model.ShowMenu}}">
-            <nav class="products-top-menu">
+            <nav slot="products/nav-bar" class="products-top-menu">
```
Comment (Mackiovello) [inline src/Products/wwwroot/Products/viewmodels/MasterPage.html]: Will this actually work when it's wrapped in a `dom-if`

File: src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html [hunk 2]
```diff
@@ -29,7 +29,6 @@ <h5 slot="products/header">
         <div style="display: flex; align-items: center; justify-content: space-between;">
             <slot name="products/header"></slot>
             <div>
-                <slot name="products/sell"></slot>
                 <slot name="products/event"></slot>
```
Comment (Mackiovello) [inline src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html]: Why isn't the timeline element included here?

Comment (Mihaiii) [inline src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html]: You mean inside this div? It was (still is) outside of this div. Should I include it here?

Comment (Mackiovello) [inline src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html]: That's my bad, I didn't see that it was below. Ignore the comment

File: src/Products/wwwroot/Products/viewmodels/MasterPage.html (hunk 1 above)
Comment (Mihaiii) [inline src/Products/wwwroot/Products/viewmodels/MasterPage.html]: I made this slot attribute because otherwise  the menu is displayed at the bottom.

Comment (Mackiovello) [inline src/Products/wwwroot/Products/viewmodels/MasterPage.html]: That's fine

**Pull request merged**